controller_manager:
  ros__parameters:
    update_rate: 100

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    torso_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    head_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    arm_left_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    arm_right_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_left_controller:
      type: joint_trajectory_controller/JointTrajectoryController

    gripper_right_controller:
      type: joint_trajectory_controller/JointTrajectoryController


# -----------------------------
# JointTrajectoryControllerë“¤
# -----------------------------
torso_controller:
  ros__parameters:
    joints: [torso_lift_joint]
    command_interfaces: [position]
    state_interfaces: [position, velocity]
    allow_partial_joints_goal: true

head_controller:
  ros__parameters:
    joints: [head_1_joint, head_2_joint]
    command_interfaces: [position]
    state_interfaces: [position, velocity]
    allow_partial_joints_goal: true

arm_left_controller:
  ros__parameters:
    joints:
      - arm_left_1_joint
      - arm_left_2_joint
      - arm_left_3_joint
      - arm_left_4_joint
      - arm_left_5_joint
      - arm_left_6_joint
      - arm_left_7_joint
    command_interfaces: [position]
    state_interfaces: [position, velocity]
    allow_partial_joints_goal: true

arm_right_controller:
  ros__parameters:
    joints:
      - arm_right_1_joint
      - arm_right_2_joint
      - arm_right_3_joint
      - arm_right_4_joint
      - arm_right_5_joint
      - arm_right_6_joint
      - arm_right_7_joint
    command_interfaces: [position]
    state_interfaces: [position, velocity]
    allow_partial_joints_goal: true

gripper_left_controller:
  ros__parameters:
    joints:
      - gripper_left_left_finger_joint
      - gripper_left_right_finger_joint
    command_interfaces: [position]
    state_interfaces: [position, velocity]
    allow_partial_joints_goal: true

gripper_right_controller:
  ros__parameters:
    joints:
      - gripper_right_left_finger_joint
      - gripper_right_right_finger_joint
    command_interfaces: [position]
    state_interfaces: [position, velocity]
    allow_partial_joints_goal: true
